spark/src/main/scala/io/delta/sharing/spark/DeltaSharingSource.scala

```diff
@@ -278,7 +377,10 @@ case class DeltaSharingSource(
    * @param indexedFiles actions list from which to generate the DataFrame.
    */
   private def createDataFrame(indexedFiles: Seq[IndexedFile]): DataFrame = {
-    val addFilesList = indexedFiles.map(_.getFileAction)
+    val addFilesList = indexedFiles.map { indexedFile =>
+      val add = indexedFile.add
+      AddFile(add.url, add.id, add.partitionValues, add.size, add.stats)
```

Comment:
should add a check that add should not be null?

Reply:
added a comment. I don't think we have to do extra check, just let it throw null pointer exception is it's actually null, which means. there's some bugs in the code.

Reply:
Should we add an assert (which will throw a more meaningful message than a null ptr exception) ?

Reply:
sg, added.